addon-api/background/MessagePasser.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Handles message passing. Mimics chrome.runtime message passing.
+ */
+
+export default class MessagePasser {
+  constructor(addonObject) {
+    this._addonId = addonObject.self.id;
+
+    this._onMessageListeners = [];
+    this._onConnectListeners = [];
+
+    let self = this;
```

Comment:
just use `this`? why do you need a `self` variable?

Reply:
Because the functions are not going to be called immediately always, so this will either be the window or undefined.